metadata/metadata_test.go

```diff
@@ -18,6 +18,10 @@ var (
 	parseErr         = errors.New("Metadata proxy could not safely parse request.")
 )
 
+func unknownQueryParameterErr(key string) error {
+	return fmt.Errorf("Unrecognized query parameter key: \"%#v\".", key)
```

Comment:
just use %q, also errors shouldn't have periods.

https://github.com/golang/go/wiki/CodeReviewComments#error-strings

Reply:
I specifically used %#v so that non-unicode characters show up as byte literals

Reply:
looks like `%#q` does the right thing

Reply:
done!